lib/flags.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+class Flags {
+  static from(...args) {
+    const values = new Map(args.map((v, i) => [v, i]));
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        if (args[0] instanceof Uint64) {
+          this.value = new Uint64(args[0]);
+        } else {
+          this.value = new Uint64(0);
+          args.forEach(arg => this.set(arg));
+        }
+      }
+
+      has(key) {
+        return values.has(key);
+      }
+
+      get(key) {
+        return values.has(key) ?
+          Uint64.shiftRight(this.value, values.get(key)).toUint32() & 1 : -1;
+      }
+
+      set(key) {
+        if (values.has(key)) {
+          this.value.or(new Uint64(1).shiftLeft(values.get(key)));
+        }
+      }
+
+      unset(key) {
+        if (values.has(key)) {
+          this.value.and(new Uint64(1).shiftLeft(values.get(key)).not());
```

Comment:
Cache all Uints (0b1, 0b10, 0b100 ...) and get needed from array? =)

Reply:
That will mean that we will duplicate Uint in the array. Also, if we are to duplicate it I'd prefer to use another `Map`. But as far as I know, `Flags` should have fast read speed meaning that data should be written on value change.

Reply:
What do you mean
> duplicate Uint in the array

?
I suggested to have an array of format `[UIn64(0b0), UIn64(0b1), UIn64(0b10), UIn64(0b100)...]`
And then use `this.value.and(arr[value.get(key)]).not()` etc.